src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,208 @@
+  certType: "SHA_1" | "SHA_256";
+  shaHash: string;
+}
+
+/**
+ * Send an API request to create a new Firebase iOS app and poll the LRO to get the new app
+ * information.
+ * @return {Promise} this function returns a promise that resolves to the new iOS app information
+ */
+export async function createIosApp(
+  projectId: string,
+  displayName: string,
+  bundleId: string
+): Promise<IosAppMetadata> {
+  const spinner = new OraWrapper(`Creating your ${clc.red("iOS")} app`);
+  spinner.start();
+
+  try {
+    const response = await api.request("POST", `/v1beta1/projects/${projectId}/iosApps`, {
+      auth: true,
+      origin: api.firebaseApiOrigin,
+      timeout: 15 * ONE_SECOND_MILLIS,
+      data: { displayName, bundleId },
+    });
+    const result = await pollOperation<any>({
+      pollerName: "Create iOS app Poller",
+      apiOrigin: api.firebaseApiOrigin,
+      apiVersion: "v1beta1",
+      operationResourceName: response.body.name /* LRO resource name */,
+    });
+    spinner.succeed();
+    const appMetadata: IosAppMetadata = {
+      appId: result.appId,
+      displayName: result.displayName,
+      bundleId: result.bundleId,
+      appPlatform: AppPlatform.IOS,
+    };
+    logPostAppCreationInformation(appMetadata);
+    return appMetadata;
+  } catch (err) {
+    spinner.fail();
+    logger.debug(err.message);
+    throw new FirebaseError(
+      `Failed to create iOS app for project ${projectId}. See firebase-debug.log for more info.`,
+      { exit: 2, original: err }
+    );
+  }
+}
+
+/**
+ * Send an API request to create a new Firebase Android app and poll the LRO to get the new app
+ * information. Optionally add a SHA certificate to the app if specified.
+ * @return {Promise} this function returns a promise that resolves to the new Android app
+ * information
+ */
+export async function createAndroidApp(
+  projectId: string,
+  displayName: string,
+  packageName: string,
+  shaCertificate?: ShaCertificate
+): Promise<AndroidAppMetadata> {
+  let spinner = new OraWrapper(`Creating your ${clc.red("Android")} app`);
+  let isProjectCreated = false;
+  spinner.start();
+
+  try {
+    const response = await api.request("POST", `/v1beta1/projects/${projectId}/androidApps`, {
+      auth: true,
+      origin: api.firebaseApiOrigin,
+      timeout: 15 * ONE_SECOND_MILLIS,
+      data: { displayName, packageName },
+    });
+    const result = await pollOperation<any>({
+      pollerName: "Create Android app Poller",
+      apiOrigin: api.firebaseApiOrigin,
+      apiVersion: "v1beta1",
+      operationResourceName: response.body.name /* LRO resource name */,
+    });
+    spinner.succeed();
+    isProjectCreated = true;
+    const appMetadata: AndroidAppMetadata = {
+      appId: result.appId,
+      displayName: result.displayName,
+      packageName: result.packageName,
+      appPlatform: AppPlatform.ANDROID,
+    };
+    logPostAppCreationInformation(appMetadata);
+    if (shaCertificate) {
```

Comment:
nit: Let's add an empty line above this `if` statement, to split up this block of code a bit.

Reply:
Done